ingest-app/src/cmr/ingest/services/email_processing.clj

```diff
@@ -0,0 +1,251 @@
+              native-id (get subscription :native-id)
+              provider-id (get subscription :provider-id)
+              [permission-check-failed permission-check-time] (check-collection-permissions
+                                                               context
+                                                               coll-id
+                                                               subscriber-id
+                                                               (get-in subscription [:extra-fields :permission-check-time])
+                                                               (get-in subscription [:extra-fields :permission-check-failed]))
+              query-string (-> (:metadata subscription)
+                               (json/decode true)
+                               :Query)
+              query-params (create-query-params query-string)
+              params1 (merge {:created-at time-constraint
+                              :collection-concept-id coll-id
+                              :token (config/echo-system-token)}
+                             query-params)
+              params2 (merge {:revision-date time-constraint
+                              :collection-concept-id coll-id
+                              :token (config/echo-system-token)}
+                             query-params)
+              gran-refs (search-gran-refs-by-collection-id context params1 params2 sub-id)
+              subscriber-filtered-gran-refs (filter-gran-refs-by-subscriber-id context gran-refs subscriber-id)]]
+    (do
+      (send-update-subscription-notification-time! context {:subscription-concept-id sub-id
+                                                            :permission-check-failed (str (true? permission-check-failed))
+                                                            :permission-check-time permission-check-time})
+      [sub-id coll-id native-id provider-id subscriber-filtered-gran-refs
+       email-address subscription permission-check-time
+       permission-check-failed])))
+
+(defn- send-subscription-emails
+  "Takes processed subscription tuples and sends out emails if applicable."
+  [context subscriber-filtered-gran-refs-list]
+  (doseq [subscriber-filtered-gran-refs-tuple subscriber-filtered-gran-refs-list
+          :let [[sub-id coll-id native-id provider-id subscriber-filtered-gran-refs
+                 email-address subscription permission-check-time
+                 permission-check-failed] subscriber-filtered-gran-refs-tuple
+                email-settings {:host (email-server-host) :port (email-server-port)}]]
+    (if (and (t/after? (t/minus (tk/now) (t/days subscription-permission-notification-expiration-days))
+                       (or (when permission-check-time
+                             (cr/from-string permission-check-time))
+                           (tk/now)))
+             permission-check-failed)
+      (try
+        (ingest/delete-concept
```

Comment:
It is odd/wrong to delete a concept in a function named `send-subscription-emails`.

Reply:
the ticket is asking for the subscription to be removed. I'm making the comments clear to this point and also writing a log entry.